lxd/api_internal.go

```diff
@@ -342,6 +349,35 @@ func internalContainerOnStart(d *Daemon, r *http.Request) response.Response {
 	return response.EmptySyncResponse
 }
 
+func internalContainerOnStartHost(d *Daemon, r *http.Request) response.Response {
+	s := d.State()
+
+	inst, err := internalContainerHookLoadFromReference(s, r)
+	if err != nil {
+		logger.Error("The start-host hook failed to load", logger.Ctx{"err": err})
+		return response.SmartError(err)
+	}
+
+	lxcPID := request.QueryParam(r, "lxc_pid")
+	if lxcPID == "" {
+		err := fmt.Errorf("No lxc_pid GET parameter was provided")
+		logger.Error("The start-host hook failed", logger.Ctx{"instance": inst.Name(), "err": err})
+		return response.BadRequest(err)
+	}
+
+	args := map[string]string{
+		"LXC_PID": lxcPID,
+	}
+
+	err = inst.OnHook(instance.HookStartHost, args)
+	if err != nil {
+		logger.Error("The start-host hook failed", logger.Ctx{"instance": inst.Name(), "err": err})
```

Comment:
I suggest making a logger that can be used both here and on line 364 by doing:

```go
l := logger.AddContext(logger.Ctx{"project": inst.Project().Name, "instance": inst.Name()})
```

Then use the `l` for anything instance related, e.g. `l.Error("The start-host hook failed", logger.Ctx{"err": err})`